CommunicationLayer/MessagingFramework/MessageDecodingHelpers.cpp

```diff
@@ -27,23 +27,47 @@
 
 namespace
 {
-   const int NUMBER_OF_BYTES_IN_FLOAT = 4;
+    const int UNION_SIZE = 4;
 
-   union FloatCharTranslator
-   {
-      float floatData;
-      char charData[NUMBER_OF_BYTES_IN_FLOAT];
-   };
+    union DataFromCharTranslator
+    {
+        unsigned short unsignedShortData;
+        float floatData;
+        char charData[MessageDecodingHelpers::numberOfBytesInData(UNION_SIZE)];
+    }
 }
 
-float MessageDecodingHelpers::getFloat(
-   const QByteArray& data, int startIndex)
+float MessageDecodingHelpers::getFloat(const QByteArray& data, int startIndex)
 {
-   FloatCharTranslator dataUnion;
-   for (int i = 0; i < NUMBER_OF_BYTES_IN_FLOAT; i++)
-   {
-      dataUnion.charData[i] = data.at(i + startIndex);
-   }
+   return getData(data, startIndex, Type::FLOAT).floatData;
+}
+
+unsigned short MessageDecodingHelpers::getUnsignedShort(const QByteArray& data, int startIndex)
+{
+    return getData(data, startIndex, TYPE::UNSIGNED_SHORT).unsignedShortData;
+}
 
-   return dataUnion.floatData;
+void MessageDecodingHelpers::getData(const QByteArray& data, int startIndex, Type type)
+{
+    DataFromCharTranslator dataUnion;
+    for (int i = 0; i < numberOfBytesInData(type); i++)
+    {
+        dataUnion.charData[i] = data.at(i = startIndex);
+    }
+    return dataUnion;
+}
+
+const int MessageDecodingHelpers::numberOfBytesInData(const Type& type)
+{
+    switch (type)
+    {
+    case FLOAT:
+        return sizeof(float);
+        break;
+    case UNSIGNED_SHORT:
+        return sizeof(unsigned short);
+        break;
+    default:
+        throw "Unhandled data type!";
+    }
```

Comment:
I know this default should never ever be run assuming the programmer knows what they're doing when they edit this file, but my gut feeling is to leave it here because I don't see how it can hurt and it could help debug issues later down the road. Should it remain or maybe QDebug() would be more appropriate? Just not sure of the standards you guys set for this sort of thing.